onecodex/viz/_functional.py

```diff
@@ -0,0 +1,169 @@
+from onecodex.lib.enums import (
+    FunctionalAnnotations,
+    FunctionalAnnotationsMetric,
+)
+from onecodex.viz._primitives import prepare_props, sort_helper
+
+
+class VizFunctionalHeatmapMixin(object):
+    def plot_functional_heatmap(
```

Comment:
I do not think it will work:
- Parameters are different already (For example: functional run y-axis can be function names or function ids). And if the future I expect them to divergent from each other even more
- They do not share much of internal pandas logic.
- Typing `samples.plot_heatmap(..., type="functional")` takes longer than `samples.plot_functional_heatmap(...)` 😀